File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int sum=0;
    public void dfs(TreeNode t,int k){
        if(t==null)return;
        k=k*10+t.val;
        if(t.left==null && t.right==null){
            sum+=k;
            return;}
        dfs(t.left,k);
        dfs(t.right,k);
    }
    
    public int sumNumbers(TreeNode root) {
        dfs(root,0);
        return sum;
    }
}

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    int ans=0;
    
    void sum(TreeNode*t,int num){
        if(!t)return;
        if(!t->left && !t->right){
            ans+=num*10+t->val;
            return;
        }
        sum(t->left,num*10+t->val);
        sum(t->right,num*10+t->val);
    }
    
    int sumNumbers(TreeNode* root) {
       
        sum(root,0);
        return ans;
    }
};